cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -1382,6 +1382,32 @@ TEST_F(TestAzuriteFileSystem, DeleteDirContentsFailureNonexistent) {
   this->TestDeleteDirContentsFailureNonexistent();
 }
 
+TEST_F(TestAzuriteFileSystem, DeleteFileSuccess) {
+  CreateFile(fs(), "abc", "data");
+  arrow::fs::AssertFileInfo(fs(), "abc", FileType::File);
+  ASSERT_OK(fs()->DeleteFile("abc"));
+  arrow::fs::AssertFileInfo(fs(), "abc", FileType::NotFound);
+}
+
+TEST_F(TestAzuriteFileSystem, DeleteFileFailureNonexistent) {
+  ASSERT_RAISES(IOError, fs()->DeleteFile("nonexistent"));
```

Comment:
This deletes a container not a blob. Could you create a container and delete an nonexistent blob in the container?

Reply:
Sure.  I have changed the test to the following:

```
TEST_F(TestAzuriteFileSystem, DeleteFileFailureNonexistent) {
  const auto container_name = PreexistingData::RandomContainerName(rng_);
  auto container_client = CreateContainer(container_name);
  const auto blob_path = ConcatAbstractPath(container_name, "someblob");
  ASSERT_RAISES(IOError, fs()->DeleteFile(blob_path));
}

```